optuna/samplers/_tpe/multi_objective_sampler.py

```diff
@@ -140,6 +140,10 @@ def __init__(
         )
         self._n_ehvi_candidates = n_ehvi_candidates
         self._mo_random_sampler = RandomSampler(seed=seed)
+        self._split_cache: Dict[int, Any] = {}
+        self._split_cache_lock = threading.Lock()
+        self._weights_below: Dict[int, Any] = {}
+        self._weights_below_lock = threading.Lock()
```

Comment:
Is it necessary to take a lock to achieve this feature? In Optuna's parallelization scheme, one running trial is tied to only one process/one thread. It would be possible to remove all the code associated with the lock.

Reply:
I have removed locks.